inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/trigger/DirectoryTrigger.java

```diff
@@ -159,9 +159,11 @@ private void registerNewDir(PathPattern entity,
                 LOGGER.info("overflow got {}", parentPath);
                 // check whether parent path is valid.
                 if (Files.isDirectory(parentPath)) {
-                    for (Iterator<Path> it = Files.list(parentPath).iterator(); it.hasNext();) {
-                        Path childPath = it.next();
-                        registerAllSubDir(entity, parentPath.resolve(childPath), tmpWatchers);
+                    try(final Stream<Path> pathStream = Files.list(parentPath)) {
```

Comment:
Please refer to the link <https://github.com/apache/incubator-inlong/discussions/3082> to set your code style, then pull the request again.